src/solution.jl

```diff
@@ -14,30 +14,33 @@ function DiscSolution(sol, semi, dimension_type::AbstractDimension)
 end
 
 #1D version of plot
-function dataplot(solution::DiscSolution{<:OneDimension}, t)
-    return PlotData1D(solution.sol(t), solution.semi)
+function dataplot(solution::DiscSolution{<:OneDimension}, solver, t)
+    return PlotData1D(solver(t), solution.semi)
 end
 
 #2D version of plot
-function dataplot(solution::DiscSolution{<:TwoDimension}, t)
-    return PlotData2D(solution.sol(t), solution.semi)
+function dataplot(solution::DiscSolution{<:TwoDimension}, solver, t)
+    return PlotData2D(solver(t), solution.semi)
 end
 
 dataplot(solution::DiscSolution{<:AbstractDimension}, t) = error("Unknown simulation type: $(typeof(t))")
 
-function plotgif(solution, tmin, tmax, kwargs...)
+function plotgif(solution::DiscSolution, tmin, tmax)#, kwargs...)
```

Comment:
Something you could do here is forward the keyword arguments to the plot command, so that the limits may be set by the user.

For example,

```julia
function plotgif(sol, tmin, tmax; kwargs...)
    # ...
        plot(pd ; kwargs...)
 end
```